app/views/flavor/_flavor_form.html.haml

```diff
@@ -0,0 +1,119 @@
+                            :name        => "vcpus",
+                            "id"         => "vcpus",
+                            'ng-model'   => "vm.flavorModel.vcpus",
+                            :required    => "",
+                            "int-greater-than-zero" => true}
+        %span.help-block{"ng-show" => "angularForm.vcpus.$error.required"}
+          = _("Required")
+        %span.help-block{"ng-show" => "angularForm.vcpus.$error.intGreaterThanZero"}
+          = _("VCPUs must be integer greater than 0")
+
+
+    .form-group{"ng-class" => "{'has-error': angularForm.disk.$error.intGreaterThanZero}"}
+      %label.col-md-2.control-label
+        = _('Disk size in GB')
+      .col-md-8
+        %input.form-control{:type        => "text",
+                            :name        => "disk",
+                            "id"         => "disk",
+                            'ng-model'   => "vm.flavorModel.disk",
+                            :required    => "",
+                            "int-greater-than-zero" => true}
+        %span.help-block{"ng-show" => "angularForm.disk.$error.required"}
+          = _("Required")
+        %span.help-block{"ng-show" => "angularForm.disk.$error.intGreaterThanZero"}
+          = _("Disk size must be integer greater than 0")
+
+    .form-group{"ng-class" => "{'has-error': angularForm.swap.$error.intGreaterOrZero}"}
+      %label.col-md-2.control-label
+        = _('Swap size in MB')
+      .col-md-8
+        %input.form-control{:type        => "text",
+                            :name        => "swap",
+                            "id"         => "swap",
+                            'ng-model'   => "vm.flavorModel.swap",
+                            :required    => "",
+                            "int-greater-or-zero" => true}
+        %span.help-block{"ng-show" => "angularForm.swap.$error.required"}
+          = _("Required")
+        %span.help-block{"ng-show" => "angularForm.swap.$error.intGreaterOrZero"}
+          = _("Swap size must be integer greater than or equal to 0")
+
+    .form-group{"ng-class" => "{'has-error': angularForm.rxtx_factor.$error.floatGreaterThanZero}"}
+      %label.col-md-2.control-label
+        = _('RXTX factor')
+      .col-md-8
+        %input.form-control{:type        => "text",
+                            :name        => "rxtx_factor",
+                            "id"         => "rxtx_factor",
+                            'ng-model'   => "vm.flavorModel.rxtx_factor",
+                            :required    => "",
+                            "float-greater-than-zero" => true}
+        %span.help-block{"ng-show" => "angularForm.rxtx_factor.$error.required"}
+          = _("Required")
+        %span.help-block{"ng-show" => "angularForm.rxtx_factor.$error.floatGreaterThanZero"}
+          = _("RXTX factor must be greater than or equal to 0")
+
+    = render :partial => "layouts/angular/x_edit_buttons_angular"
```

Comment:
You want `generic_form_buttons` here instead .. those x_edit are for non-controllerAs controllers.